src/user/user.c

```diff
@@ -42,28 +47,23 @@ int main(int ac, char **argv)
 		return 0;
 	}
 
-	// get the fd for the map created in the BPF program
-	map_fd = bpf_map__fd(skel->maps.output_map);
-
-	// count the total number of processes executed on the CPUs each second
-	while (1) {
-		/*
-		 * Search for the element with key = 0 in the map.
-		 * values will point to output_map[key]
-		 */
-		bpf_map_lookup_elem(map_fd, &key, executions_nb_on_cpu);
-
-		// reset the counter to avoid counting duplicates
-		total_executions = 0;
-
-		// accumulate execution number for all CPUs
-		for (int i = 0; i < cpus_nb; i++) {
-			total_executions += executions_nb_on_cpu[i];
-		}
+	ringbuffer_fd = bpf_map__fd(skel->maps.ringbuf);
+	if (ringbuffer_fd < 0) {
+		printf("Error accessing the ringbuffer.\n");
+		exec__destroy(skel);
+		return 0;
+	}
 
-		printf("There were %d processes executed on %d CPUs\n", total_executions, cpus_nb);
-		sleep(1);
+	ringbuffer = ring_buffer__new(ringbuffer_fd, process_sample, NULL, NULL);
+	if (!ringbuffer) {
+		printf("Error creating the ringbufffer.\n");
+		exec__destroy(skel);
+		return 0;
 	}
+	
+	while (1)
```

Comment:
If you have a comment in your loop, please use brackets, otherwise it gets very confusing to read.

You can omit the bracket but only when there is a single indented line after it.

Reply:
ah, ok :D